src/components/data/itemgrid/itemGrid.tsx

```diff
@@ -0,0 +1,279 @@
+import React from "react";
+
+import {
+  AttributeData,
+  DEFAULT_URL_FIELDS,
+  FieldSet,
+  field2Title,
+  formatMessage,
+} from "../../../lib";
+import { Button, ButtonLink, ButtonLinkProps, ButtonProps } from "../../button";
+import { Column, Grid } from "../../layout";
+import { Body, H1, H2, H3, P } from "../../typography";
+import "./itemgrid.scss";
+
+export type ItemGridProps = Omit<
+  React.ComponentProps<"div">,
+  "title" | "onClick"
+> & {
+  /** A title for the datagrid. */
+  title?: React.ReactNode;
+
+  /** A `Function` that is used to create the preview for an object. */
+  renderPreview?: (attributeData: AttributeData) => React.ReactNode;
+
+  /**
+   *  A `string[]` containing the fields which should be used to detect the url
+   *  of a row. Fields specified in this object won't be rendered, instead: the
+   *  first (non url) field is rendered as link (`A`) with `href` set to the
+   *  resolved url of the row.
+   */
+  urlFields?: string[];
+
+  buttonLinkProps?: ButtonLinkProps;
+  buttonProps?: ButtonProps;
+
+  /**
+   * Gets called when a button is clicked.
+   * @param event
+   * @param attributeData
+   */
+  onClick?: (event: React.MouseEvent, attributeData: AttributeData) => void;
+} & ItemGridConfigurationProps;
+
+export type ItemGridConfigurationProps =
+  | ItemGridFieldsetProps
+  | ItemGridGroupByProps;
+
+/**
+ * WHen using `fieldsets` / `objectLists` to define sections. The length of `fieldSets` and `objectLists` should match.
+ */
+export type ItemGridFieldsetProps = {
+  /** The fieldsets to render. */
+  fieldsets: FieldSet[];
+
+  /** The object lists to show, each index should match `fieldsets` index. */
+  objectLists: AttributeData[][];
+
+  groupBy?: undefined;
```

Comment:
Could we add some docs to type definitions that don't in this file like these two?